packages/astro/test/units/dev/head-injection.test.js

```diff
@@ -0,0 +1,160 @@
+import { expect } from 'chai';
+import * as cheerio from 'cheerio';
+
+import { runInContainer } from '../../../dist/core/dev/index.js';
+import { createFs, createRequestAndResponse } from '../test-utils.js';
+
+const root = new URL('../../fixtures/alias/', import.meta.url);
+
+describe('head injection', () => {
```

Comment:
Couldn't quite tell from these test cases, but will head injection support inline styles alongside standard stylesheets? I know that `renderEntry` needs to handle both inline styles and links from the render result today, since MDX outputs as inline styles in dev, stylesheets in production.

Reply:
This makes it possible to create a component that does head injection. So `renderEntry` will need to use the right APIs, such as the `createComponent` object form, to return the CSS it needs.

So if I'm understanding you right, that work will be done in the renderEntry PR.

Reply:
@matthewp got it. I'm asking since I know a component may want to inject an inline `style` or a `link` tag. It sounds like this API gives you the freedom to inject _any_ HTML element + element children that you want?

Reply:
Right, you can return any string or iterable, essentially the same thing you can do in a template. This PR contains a helper for rendering a stylesheet: https://github.com/withastro/astro/pull/5511/files#diff-a2d99b846ae3297fe01c808e11546d92b906ffafdbc17134bca73edf276fd3b5R17

I'll probably add more helpers in the other PR.